src/mainwindow.cpp

```diff
@@ -701,6 +701,18 @@ void MainWindow::compileAndRun()
     compile();
 }
 
+void MainWindow::submitSolution()
+{
+    LOG_INFO("Requesting submit solution");
+    if (submitToCodeforces)
```

Comment:
```suggestion
    if (canSubmitSolution())
```

Reuse codes when possible.